contracts/src/instances/aave/AaveBase.sol

```diff
@@ -0,0 +1,207 @@
+        _vaultSharesToken.transferFrom(
+            msg.sender,
+            address(this),
+            // NOTE: The AToken interface transfers in base, so we have to
+            // convert the share amount to a base amount.
+            _convertToBase(_shareAmount)
+        );
+    }
+
+    /// @dev Process a withdrawal in base and send the proceeds to the
+    ///      destination.
+    /// @param _shareAmount The amount of vault shares to withdraw.
+    /// @param _destination The destination of the withdrawal.
+    /// @return amountWithdrawn The amount of base withdrawn.
+    function _withdrawWithBase(
+        uint256 _shareAmount,
+        address _destination,
+        bytes calldata // unused
+    ) internal override returns (uint256 amountWithdrawn) {
+        // Withdraw assets from the Aave vault to the destination.
+        amountWithdrawn = _vault.withdraw(
+            address(_baseToken), // asset
+            // NOTE: Withdrawals are processed in base, so we have to convert
+            // the share amount to a base amount.
+            _convertToBase(_shareAmount), // amount
+            _destination // onBehalfOf
+        );
+
+        return amountWithdrawn;
+    }
+
+    /// @dev Process a withdrawal in vault shares and send the proceeds to the
+    ///      destination.
+    /// @param _shareAmount The amount of vault shares to withdraw.
+    /// @param _destination The destination of the withdrawal.
+    function _withdrawWithShares(
+        uint256 _shareAmount,
+        address _destination,
+        bytes calldata // unused
+    ) internal override {
+        // NOTE: We don't need to use `safeTransfer` since ATokens are
+        // standard-compliant.
+        //
+        // Transfer vault shares to the destination.
+        _vaultSharesToken.transfer(
+            _destination,
+            // NOTE: The AToken interface transfers in base, so we have to
+            // convert the share amount to a base amount.
+            _convertToBase(_shareAmount)
+        );
+    }
+
+    /// @dev Convert an amount of vault shares to an amount of base.
+    /// @param _shareAmount The vault shares amount.
+    /// @return The base amount.
+    function _convertToBase(
+        uint256 _shareAmount
+    ) internal view override returns (uint256) {
+        // Aave's AToken accounting calls shares "scaled tokens." We can convert
+        // from scaled tokens to aTokens with the formula:
+        //
+        // aToken = scaledToken.rayMul(POOL.getReserveNormalizedIncome(_underlyingAsset))
+        //
+        // `rayMul` computes a 27 decimal fixed point multiplication and
+        // `_underlyingAsset` is the base token address.
+        //
+        // NOTE: We use `mulDivDown` with 27 decimals of precision to compute
+        // the calculation to ensure that we are always rounding down since
+        // `rayDiv` will round up in some cases.
+        return _shareAmount.mulDivDown(_getReserveNormalizedIncome(), 1e27);
```

Comment:
kind of annoying that this can't be used in the deploerCoordinator

Reply:
I'm glad you said this. The conversion functions are nice.